src/main/java/de/tum/cit/aet/artemis/quiz/service/QuizQuestionProgressService.java

```diff
@@ -227,18 +240,21 @@ public void saveProgressFromTraining(QuizQuestion question, Long userId, Submitt
         QuizQuestionProgress existingProgress = quizQuestionProgressRepository.findByUserIdAndQuizQuestionId(userId, question.getId()).orElse(new QuizQuestionProgress());
         QuizQuestionProgressData data = existingProgress.getProgressJson() != null ? existingProgress.getProgressJson() : new QuizQuestionProgressData();
 
-        existingProgress.setLastAnsweredAt(answeredAt);
-        existingProgress.setQuizQuestionId(question.getId());
-        existingProgress.setUserId(userId);
-        double score = question.getPoints() > 0 ? answer.getScoreInPoints() / question.getPoints() : 0.0;
-        updateProgressWithNewAttempt(data, score, answeredAt);
-        updateProgressCalculations(data, score, existingProgress);
-        existingProgress.setProgressJson(data);
-        try {
-            quizQuestionProgressRepository.save(existingProgress);
-        }
-        catch (DataIntegrityViolationException e) {
-            updateExistingProgress(userId, question, data, answeredAt);
+        ZonedDateTime dueDate = data.getDueDate() != null ? data.getDueDate() : null;
```

Comment:
```suggestion
        ZonedDateTime dueDate = data.getDueDate();
```
What kind of gymnastics is this? :P